### PR TITLE
[Fix] Fix pointpillars bug on kitti and performance upgrades

### PR DESCRIPTION
**Pull request by ZCMax**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
Fix pointpillars bug on kitti.

## Modification
change the config for the anchor generator range.

## BC-breaking (Optional)
Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)
If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by Tai-Wang — APPROVED** [review]

**Comment by ZCMax** [comment]
The bug fix and performance improvements will be both in this PR

File: mmdet3d/models/voxel_encoders/pillar_encoder.py [hunk 1]
```diff
@@ -47,14 +46,14 @@ def __init__(self,
                  point_cloud_range=(0, -40, -3, 70.4, 40, 1),
                  norm_cfg=dict(type='BN1d', eps=1e-3, momentum=0.01),
                  mode='max',
-                 legacy=True):
```
Comment (ZwwWayne) [inline mmdet3d/models/voxel_encoders/pillar_encoder.py:50]: This changes default behavior. Should we avoid BC-breaking?

Comment (ZCMax) [reply mmdet3d/models/voxel_encoders/pillar_encoder.py:50]: Double check, there is no need for modifications here.

**Comment by ZCMax** [comment]
This PR will be closed, and it will be divided into two PRs. One for bug fix on master branch, and the other for performance upgrade on dev branch.

**Pull request closed without merging**